_gsocproposals/2018/proposal_DIANAHEPanalysisfunctions.md

```diff
@@ -0,0 +1,46 @@
+---
+title: DIANA-HEP/analysisfunctions - Translate HEP analysis functions to modern paradigms
+layout: gsoc_proposal
+project: DIANA-HEP
+year: 2018
+organization:
+  - DIANA-HEP
+  - Princeton
+---
+
+# Description
+
+High Energy Physics (HEP) analysis scripts are typically written in a straightforward, procedural style, but algorithms written in this style are not always performant. Slow analysis scripts do more than slow down research— they break the concentration of the analyst. HEP can benefit from Spark-style analysis pipelining, which makes it easier to parallelize algorithms, as well as contiguous-memory and vectorization techniques from High Performance Computing (HPC). While tools to provide these features are well in development, we also need common analysis algorithms to be cast in these forms to take advantage of the new libraries. The goal of this project is to reimplement some real functions and demonstrate the improvements possible with functional, pipelined, and vectorized approaches.
```

Comment:
i'm still a bit hung up on scripts:) - i think we would benefit from a word on what scripts can do - eg "As in many data sciences, analysis scripts do anything from data reduction, to data combination, up to preparing data for visualization."

but that got pretty complicated

Reply:
Can you change it directly?

Reply:
i assume i can not change your pr - maybe I can make a PR to your branch for you to accept- but i wasn't wanting to change something that didn't improve things...

Reply:
Uh... yeah, I think you're right. Since this is a fork-to-PR, you'd have to fork my fork and PR it to me for me to PR to hep-sf. Too complicated to be worth trying. (It would be different if this were a PR within a repository.) If you do come up with a better wording, let me know and I'll change it right away. As long as I change it in my fork before this PR is accepted, the change would be carried along.

Reply:
super - Thanks @jpivarski